src/apps/revert-finance/arbitrum/revert-finance.balance-fetcher.ts

```diff
@@ -17,34 +21,64 @@ const network = Network.ARBITRUM_MAINNET;
 
 @Register.BalanceFetcher(REVERT_FINANCE_DEFINITION.id, network)
 export class ArbitrumRevertFinanceBalanceFetcher implements BalanceFetcher {
-  constructor(@Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit) {}
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(UniswapV2ContractFactory) protected readonly uniswapV2ContractFactory: UniswapV2ContractFactory,
+    @Inject(UniswapV3LiquidityTokenHelper)
+    private readonly uniswapV3LiquidityTokenHelper: UniswapV3LiquidityTokenHelper,
+  ) {}
 
-  async getAccumulatedCompoundorRewards(address: string) {
+  async getCompoundorAccountBalances(address: string) {
     const graphHelper = this.appToolkit.helpers.theGraphHelper;
-    const data = await graphHelper.requestGraph<CompoundorUserPosition>({
+    const data = await graphHelper.requestGraph<CompoundorAccountBalances>({
       endpoint: generateGraphUrlForNetwork(network),
       query: accountBalancesQuery,
       variables: { address: getAddress(address) },
     });
     if (!data) return [];
     const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
-    const rewards: Array<ContractPositionBalance> = [];
+    const accountBalances: Array<ContractPositionBalance> = [];
     data.accountBalances.map(({ token, balance }) => {
       const existingToken = baseTokens.find(item => item.address === token)!;
       if (!token) return [];
+      accountBalances.push(getCompoundorContractPosition(network, existingToken, balance));
+    });
+    return accountBalances;
+  }
 
-      rewards.push(getCompoundorContractPosition(network, existingToken, balance));
+  async getCompoundingAccountTokens(address: string) {
+    const graphHelper = this.appToolkit.helpers.theGraphHelper;
+    const data = await graphHelper.requestGraph<CompoundingAccountTokens>({
+      endpoint: generateGraphUrlForNetwork(network),
+      query: accountCompoundingTokensQuery,
+      variables: { address: getAddress(address) },
     });
-    return rewards;
+    if (!data) return [];
+    const compoundingBalances: Array<AppTokenPositionBalance> = [];
+    await Promise.all(
+      data.tokens.map(async ({ id }) => {
+        const uniV3Token = await this.uniswapV3LiquidityTokenHelper.getLiquidityToken({ positionId: id, network });
```

Comment:
For optimization, pass in `context: { multicall, baseTokens }` here as well.

Reply:
Resolved for each network
https://github.com/Zapper-fi/studio/pull/1112/files#diff-5fe2b7664cc1b3e3684b59d08e9f963b4a6b43141a4be324f22db5fb904f8f86R57